lib/modules/manager/terragrunt/modules.ts

```diff
@@ -68,18 +68,34 @@ export function analyseTerragruntModule(
     dep.currentValue = githubRefMatch.groups.tag;
     dep.datasource = GithubTagsDatasource.id;
   } else if (gitTagsRefMatch?.groups) {
-    dep.depType = 'gitTags';
-    if (gitTagsRefMatch.groups.path.includes('//')) {
+    const { url, host, path, tag } = gitTagsRefMatch.groups;
+    const containsSubDirectory = path.includes('//');
+    if (containsSubDirectory) {
       logger.debug('Terragrunt module contains subdirectory');
-      dep.depName = gitTagsRefMatch.groups.path.split('//')[0];
-      const tempLookupName = gitTagsRefMatch.groups.url.split('//');
-      dep.packageName = tempLookupName[0] + '//' + tempLookupName[1];
+    }
+    dep.depType = 'gitTags';
+    // We don't want to have .git or subdirectory in the depName
+    dep.depName = host + '/' + path.split('//')[0].replace('.git', '');
+    dep.currentValue = tag;
+    dep.datasource = detectGitTagDatasource(url);
+    if (
+      [
+        GitlabTagsDatasource.id,
+        BitbucketTagsDatasource.id,
+        GiteaTagsDatasource.id,
+      ].includes(dep.datasource)
+    ) {
```

Comment:
why not the other way around to reduce the condition?

Reply:
Makes sense :+1: